docker/pulumi/Dockerfile

```diff
@@ -39,7 +39,7 @@ RUN apt-get update -y && \
   rm -rf aws && \
   rm awscliv2.zip && \
   # Add additional apt repos all at once
-  echo "deb https://deb.nodesource.com/node_16.x $(lsb_release -cs) main"                         | tee /etc/apt/sources.list.d/node.list             && \
+  echo "deb https://deb.nodesource.com/node_18.x $(lsb_release -cs) main"                         | tee /etc/apt/sources.list.d/node.list             && \
```

Comment:
how do we make sure we don't get caught behind again? looks like Node 20 is [becoming LTS starting October 24th](https://github.com/nodejs/release#release-schedule:~:text=2023%2D04%2D18-,2023%2D10%2D24,-2024%2D10%2D22)

Reply:
Opened https://github.com/pulumi/pulumi-docker-containers/issues/158 to track.